lib/api/routes/quicklinks/discord_cdn.ex

```diff
@@ -1,46 +1,40 @@
 defmodule Lanyard.Api.Quicklinks.DiscordCdn do
-  alias Lanyard.Api.Util
+  @moduledoc """
+  A module for proxying Discord CDN images.
+  """
 
+  alias Lanyard.Api.Util
   import Plug.Conn
 
   @discord_cdn "https://cdn.discordapp.com"
 
+  @spec proxy_image(Plug.Conn.t()) :: Plug.Conn.t()
   def proxy_image(conn) do
     [user_id, file_type] =
       conn.request_path
       |> String.split("/")
       |> Enum.at(1)
       |> String.split(".")
-
+    
     presence = Lanyard.Presence.get_pretty_presence(user_id)
 
     case presence do
-      {:ok, p} ->
-        {:ok, %HTTPoison.Response{body: b, headers: h, status_code: status_code}} =
-          get_proxied_avatar(
-            user_id,
-            p.discord_user.avatar,
-            p.discord_user.discriminator,
-            file_type
-          )
-
-        conn
-        |> merge_resp_headers(h)
-        |> send_resp(status_code, b)
-
-      error ->
-        Util.respond(conn, error)
-    end
-  end
+      {:ok, %{discord_user: %{avatar: avatar}}} when is_binary(avatar) ->
+        url = construct_avatar_url(user_id, avatar, file_type)
+        get_proxied_image(conn, url)
 
-  defp get_proxied_avatar(id, avatar, _discriminator, file_type) when is_binary(avatar) do
-    constructed_cdn_url = "#{@discord_cdn}/avatars/#{id}/#{avatar}.#{file_type}?size=1024"
+      {:ok, _} ->
+        Util.respond(conn, :not_found)
 
-    HTTPoison.get(constructed_cdn_url)
+      _ ->
+        Util.respond(conn, presence)
+    end
   end
 
-  defp get_proxied_avatar(_id, avatar, discriminator, _file_type) when is_nil(avatar) do
-    mod = Integer.mod(String.to_integer(discriminator), 5)
```

Comment:
Why remove the fallback image? We want to respond with a Discord avatar no matter what